event-adapter/pkg/provider/resource_lister.go

```diff
@@ -17,31 +17,36 @@ limitations under the License.
 package provider
 
 import (
-	"k8s.io/apiserver/pkg/endpoints/handlers"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apiserver/pkg/endpoints/handlers"
 )
 
-type customMetricsResourceLister struct {
-	provider CustomMetricsProvider
+type eventsResourceLister struct {
+	provider EventsProvider
 }
 
-func NewResourceLister(provider CustomMetricsProvider) handlers.APIResourceLister {
-	return &customMetricsResourceLister{
+// NewResourceLister creates a new APIResourceLister for provided EventsProvider
+func NewResourceLister(provider EventsProvider) handlers.APIResourceLister {
+	return &eventsResourceLister{
 		provider: provider,
 	}
 }
 
-func (l *customMetricsResourceLister) ListAPIResources() []metav1.APIResource {
-	metrics := l.provider.ListAllMetrics()
-	resources := make([]metav1.APIResource, len(metrics))
-
-	for i, metric := range metrics {
-		resources[i] = metav1.APIResource{
-			Name: metric.GroupResource.String()+"/"+metric.Metric,
-			Namespaced: metric.Namespaced,
-			Kind: "MetricValueList",
-			Verbs: metav1.Verbs{"get"}, // TODO: support "watch"
-		}
+// ListAPIResources lists the supported API
+func (l *eventsResourceLister) ListAPIResources() []metav1.APIResource {
+	resources := make([]metav1.APIResource, 2)
+	resources[0] = metav1.APIResource{
+		Name:       "namespace/{namespaces}/events/{eventName}",
```

Comment:
I think it should be "namespaces/{namespaceName}/events/{eventName}", same below.